### ISSUE TITLE
Running docker-compose setup not working. Throwing an invalid password error message

### ISSUE THREAD

**Issue by asadasivan** [DefectDojo/django-DefectDojo#2082]
**Bug description**
Tried to install deject-dojo using container approach i.e. by using docker-compose but for some reason, the password provided doesn't work. 

**Steps to reproduce**
1. git clone https://github.com/DefectDojo/django-DefectDojo.git
2. cd django-DefectDojo/
3. docker-compose up
4. Open a browser and go to http://10.223.227.51:8080/login?next=/
5. Enter the following creds:
    username: admin
    password: ********************
6. scroll the docker-compose std out and you will see a section like below
    initializer_1   | Initializing.
   initializer_1   | Admin user: admin
   initializer_1   | Admin password: ********************
 initializer_1   | Waiting for database to be reachable 
7. Reasoning:
    You will get an error: " Please enter a correct username and password. Note that both fields may be case-sensitive." 
8. **Expected behavior**
The credentials should work. I think the sql is not creating the user or changing the password

**Deployment method** *(select with an `X`)*
- [ ] Kubernetes
- [X] Docker
- [ ] setup.bash / legacy-setup.bash

**Environment information**
 - System Info:
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.4 LTS
Release:	18.04
Codename:	bionic

- docker-compose version:
docker-compose version 1.25.2, build 698e2846
docker-py version: 4.1.0
CPython version: 3.7.5
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019
- DefectDojo version: 1.5.4
 - DefectDojo Commit Message: 64d5f348: Merge pull request #2032 from DefectDojo/update-contributing-master [2020-03-12 22:41:42 +0100]

**Sample scan files** (optional)
If applicable, add sample scan files to help reproduce your problem.

**Screenshots** (optional)

**Console logs** (optional)
If applicable, add console logs to help explain your problem.

**Additional context** (optional)

**Comment by TheUnriashol** [comment]
I also ran across this issue, it is without doubt a bug but there is a workaround.

First you will need the MySQL password that gets generated. You can get this by executing the following.
```
docker logs $MYSQL_CONTAINER > ./msql_log.txt
```
Just make sure to change $MYSQL_CONTAINER with the container id which you can find by executing the following.
```
docker ps
```
After you have done that you can go to the generated txt and there you will find the MySQL password.

Next just simply execute the following commands to update the admin user password to admin. Make sure to replace the XXXXX with the password you obtained above and also replace the $MYSQL_CONTAINER with the container id you also obtained above.

```
docker exec -it $MYSQL_CONTAINER bash
mysql -uroot -pXXXXX
USE defectdojo
UPDATE auth_user SET password='pbkdf2_sha256$36000$sT96yObJtsFk$F9YAJimsQqBXnff/QGLNTv100qhCNl/23hoBuNtSNZU=' WHERE username='admin';
```

Now you can login using the username `admin` and the password `admin`

SOURCE: https://github.com/DefectDojo/django-DefectDojo/issues/642#issuecomment-582857997

**Comment by asadasivan** [comment]
@TheUnriashol I tried the docker logs command to get the mysql password and the file is empty. Checking the console output, I see the following:
`mysql_1         | 2020-03-20T04:24:19.237041Z 0 [Note] Event Scheduler: Loaded 0 events
mysql_1         | 2020-03-20T04:24:19.237779Z 0 [Note] mysqld: ready for connections.
mysql_1         | Version: '5.7.27'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server (GPL)
mysql_1         | 2020-03-20T04:24:20.250532Z 2 [Note] Got an error reading communication packets
mysql_1         | 2020-03-20T04:24:21.281468Z 3 [Note] Got an error reading communication packets
mysql_1         | 2020-03-20T04:24:23.666606Z 5 [Note] Got an error reading communication packets
uwsgi_1         | wait-for-it.sh: waiting 30 seconds for mysql:3306
uwsgi_1         | wait-for-it.sh: mysql:3306 is available after 0 seconds`

**Comment by TheUnriashol** [comment]
@asadasivan 😨 Yep, you are right. The workaround doesn't work anymore. I just used this exact method about two weeks ago and it worked. No idea what happened since then. I will keep on trying some stuff, if I figure something out I will update this issue.

**Comment by TheUnriashol** [comment]
@asadasivan After testing some more I found a workaround for the workaround 😉. I know seems silly but it works. In summary for some reason the log of the MySQL container doesn't output everything it contains. So the only way to force it to show the beginning of the log is to change the volume in the docker-compose.yml. I am not sure if this has any effect on other stuff but as far as I can tell it works just fine.

Here a short video of how you can get the MySQL Root Password: https://youtu.be/wTtRJ6XyhtU

After you have that password make sure you don't shutdown the containers or else the log file wont show fully again and the password will regenerate on each startup.

Now you can continue with the following steps:
```
docker exec -it $MYSQL_CONTAINER bash
mysql -uroot -pXXXXX
USE defectdojo
UPDATE auth_user SET password='pbkdf2_sha256$36000$sT96yObJtsFk$F9YAJimsQqBXnff/QGLNTv100qhCNl/23hoBuNtSNZU=' WHERE username='admin';
```

After you update the user with the admin password you should be able to login and you can shutdown the containers as you please.

**Comment by madchap** [comment]
Please make sure to try from scratch, e.g:
* `docker-compose rm -vf`

Find out the mysql volume used and delete it. This is crucial to really start fresh on this:
* `docker volume ls |grep dojo`
* `docker volume rm <whatever_the_name_returned_in_previous_command>`

Having done that, and given the state of `dev` is very volatile, you probably want to pull the latest changes and rebuild the images with `docker-compose build`.

**Comment by madchap** [comment]
Alternatively also, if your stack is OK, If you want to create another superuser, you do not need to go through any mysql crazy command. You can just use:

`docker-compose exec uwsgi /bin/bash -c 'python manage.py createsuperuser'`

There is also a command to simply change the password, but you'd have to make sure that the docker image user uid matches yours or you'd get an error (at least that's my own workaround, not related to this issue, however).

**Comment by valentijnscholten** [comment]
Should be fixed by #2084. The docs should make it more clear that you need to grep the password from the *first run*.

**Comment by asadasivan** [comment]
That worked. Here is a summary of the steps performed to workaround the issue:
1. Fresh Install or cleanup:
```
   a. docker-compose down
   b. docker-compose rm -vf
```
2. Delete mysql volume to start fresh:
    ```
    a. docker volume ls |grep dojo
    b. docker volume rm <whatever_the_name_returned_in_previous_command>
```
3. Clone the repo and build and finally run the container
```
   a. git clone https://github.com/DefectDojo/django-DefectDojo.git
   b. cd django-DefectDojo/
   c. docker-compose build. 
   d. docker-compose up -d
```
6. Get the mysql root password:
```
    a. docker ps to get the mysql docker container id
    b. Get root password: docker logs 74e40d3e6781 > mysqllog.txt
    c. cat mysqllog.txt where you will find a line:
        GENERATED ROOT PASSWORD: xxxxxxxx
```
7. Change admin password:
```
    a. docker exec -it $MYSQL_CONTAINER bash
    b. mysql -uroot -pxxxxxxxx
    c. USE defectdojo
    d. UPDATE auth_user SET password='pbkdf2_sha256$36000$sT96yObJtsFk$F9YAJimsQqBXnff/QGLNTv100qhCNl/23hoBuNtSNZU=' WHERE username='admin';
```
8. Go to http://<ip>:8080 and login with the following credentials:
```
    username: admin
    password: admin
```
9. Change the password
